socorro/external/postgresql/connection_context.py

```diff
@@ -2,15 +2,32 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+import os
 import socket
 import contextlib
 import psycopg2
 import psycopg2.extensions
+from urlparse import urlparse
 
-from configman.config_manager import RequiredConfig
-from configman import Namespace
+from configman import RequiredConfig, Namespace
+from configman.memoize import memoize
 
+#------------------------------------------------------------------------------
+# we're going to end up parsing the same url several times, there's no need to
+# actually do it more than once, so we wrap the method with a memoizer to speed
+# things along.
+@memoize()
```

Comment:
Rewriting the standard lib `urlparse` just to put a memoize on it gains very little in my (stringent) opinion. It's a I/O-less operation that takes virtually no time at all and swapping `parse_url` for `urlparse` isn't winning us much. It just adds moving parts and future-developer-questionmarks. :)

Strangely, now that I look at your rewrite from Selena I don't see multiple places where it converts the URL into a dict (or DotDict like thing). What I wanted was to only see the `urlparse.urlparse` used in one single place in a neat little function in some `util.py` somewhere.